### PR TITLE
deprecation(components/lookup): lookup component's `data` input has been deprecated and consumers should use the `searchAsync` event instead

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**
[AB#3002583](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3002583)

### CODE REVIEW

File: libs/components/lookup/src/lib/modules/lookup/lookup-autocomplete-adapter.ts [hunk 1]
```diff
@@ -57,9 +58,8 @@ export class SkyLookupAutocompleteAdapter {
    * The function to dynamically manage the data source when users
    * change the text in the lookup field. The search function must return
    * an array or a promise of an array. The `search` property is particularly
-   * useful when the data source does not live in the source code. If the
-   * search requires calling a remote data source, use `searchAsync` instead of
-   * `search`.
+   * useful when the data source does not live in the source code.
+   * @deprecated The `searchAsync` event emitter and callback to provide searched data to the lookup component.
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/lookup/src/lib/modules/lookup/lookup-autocomplete-adapter.ts]: Should these comments be `@deprecated Use the searchAsync...` instead?

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2617/storybook/)

Component Storybooks:

- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2617/storybooks/ag-grid-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2617/storybooks/data-manager-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2617/storybooks/lookup-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2617/storybooks/phone-field-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2617/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2617/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2617/playground/)

File: libs/components/lookup/src/lib/modules/lookup/lookup.component.ts [hunk 2]
```diff
@@ -95,6 +95,7 @@ export class SkyLookupComponent
    * enter text. You can specify static data such as an array of objects, or
    * you can pull data from a database.
    * @default []
+   * @deprecated The `searchAsync` event emitter and callback to provide data to the lookup component.
```
Comment (Blackbaud-CoreyArcher) [inline libs/components/lookup/src/lib/modules/lookup/lookup.component.ts]: Same as above

File: libs/components/lookup/src/lib/modules/lookup/lookup-autocomplete-adapter.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/lookup/src/lib/modules/lookup/lookup-autocomplete-adapter.ts]: Good catch. Fixed.

File: libs/components/lookup/src/lib/modules/lookup/lookup.component.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply libs/components/lookup/src/lib/modules/lookup/lookup.component.ts]: Good catch. Fixed.

**Review by Blackbaud-CoreyArcher — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/2618

**Pull request merged**